src/common/components/repository-row/dropdowns/register-name-dropdown.js

```diff
@@ -129,7 +194,7 @@ const ActionButtons = (props) => {
       snapName === '' ||
       !!registerNameStatus.error
     );
-    actionText = 'Register name';
+    actionText = !registeredName ? 'Register name' : 'Register new name';
```

Comment:
As above, I usually find this kind of conditional easier to read when the positive case comes first.